mitodl/open-discussions
# Create and update course documents in Elasticsearch

Depends on and includes code from #1753

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #1717
Closes #1718

#### What's this PR do?
- Adds ES serializer and mapping for courses
- Creates a course index

#### How should this be manually tested?
First, import some courses (ask me or devops for values of `EDX_API_URL`, `EDX_API_CLIENT_ID`, `EDX_API_CLIENT_SECRET`):
```python
from course_catalog.tasks import get_edx_data
get_edx_data()
```

Then run `docker-compose run web python manage.py recreate_index`

Post to `http://localhost:9101/discussions_local_all_default/_search` with `{"query": {"bool": {"must": [{"match": {"object_type": "course"}}]}}}`, you should get some results.
